galaxy_ng/tests/integration/api/test_ui_paths.py

```diff
@@ -28,10 +29,31 @@
     schema_user,
 )
 from ..utils import UIClient, generate_unused_namespace, get_client
+from .rbac_actions.utils import ensure_test_container_is_pulled, podman_push
+
+from orionutils.generator import randstr
 
 REGEX_403 = r"HTTP Code: 403"
 
 
+def wait_for_task(uclient, task):
```

Comment:
the `wait_for_state` that I'm using is using the `UIClient` rather than the normal client and is also validating the task schema and asserting on the responses status code.  I figured instead of having inline in the registry test it would be good to be able to reuse it for additional UI API tests.

Reply:
good idea to break it out for reuse

since we want it for reuse, it should probably live in utils, maybe https://github.com/ansible/galaxy_ng/blob/master/galaxy_ng/tests/integration/utils/tasks.py in a wait_for_task_ui_client method?

i feel like checking json validation in a method that waits for task violates the single responsibility principle. We may want that json validation in a separate test that checks the task endpoint... as long as it ran after other tests, there would be at least one completed task it could get.

Reply:
:+1: I'll make a note to validate the task details schema when updating the tests for Pulp endpoints that galaxy_ng uses.